### ISSUE TITLE
Docker for Mac net=host issue

### ISSUE THREAD

**Issue by bomeara** [hpchud/vcc-torque#1]
Thanks for the work on VCC -- it's the perfect thing for my use case (a couple of dozen computers I want to network for a cluster). Unfortunately (for this) they're Macs, where --net=host has [issues](https://github.com/docker/for-mac/issues/68). Should I pass `-p 2379:2379 -p 4001:4001` to both discovery and headnode, should I be doing this through docker-machine instead of Docker for Mac, etc.? I've played a bit with various options but advice would be helpful. Note I'm still a bit of a noob when it comes to docker, so it's possible I'm just being stupid.

**Comment by joshiggins** [comment]
Ah! I hadn't considered this. This will be a problem on Windows as well, because `host` really means the Linux VM running Docker as pointed out in the issue you linked, so its not very intuitive at all......

In addition to the discovery ports 2379 and 4001, Torque uses 15001-15004 but the worker nodes will also need to [bind to privileged ports](http://docs.adaptivecomputing.com/8-1-3/basic/installGuide/Content/topics/hpcSuiteInstall/rpm/1-installing/configuringTorque.htm#open) for authentication that are not determined in advance (just anything < 1024). So we can't use `-p xxxx:xxxx` to forward them, we really need `--net=host`.

To get around this problem, I've been using [Weave Net](https://www.weave.works/docs/net/latest/using-weave/) to create an overlay network between hosts. This makes it so that containers running on Docker on different hosts can see each other as though they are on the same network without any port forwarding, and then you don't need `--net=host` at all.

I'll try and get hold of a Mac and write up some quick directions, I think that Docker Machine might be an easier choice if we need to use Weave Net.

**Comment by bomeara** [comment]
Thanks, @joshiggins. Let me know if  you have trouble getting a Mac to log in on -- I could give you a remote login.

**Comment by joshiggins** [comment]
@bomeara thanks! I managed to get some Macs to experiment with. Just putting together some docs now.

**Comment by joshiggins** [comment]
[Documentation is in progress](https://github.com/hpchud/vccjs/wiki/Running-VCC-container-under-Docker-Toolbox-on-Mac-or-Windows) let me know how you get on :smile_cat:

This workaround doesn't use Weave Net because that was getting quite complicated. This way we still use `--net=host` but create a bridged network adapter for the VM. The annoying thing either way is that you need to assign IPs manually to each Docker VM - the whole point of the discovery was to require only one piece of information to find the rest of the cluster.

**Comment by joshiggins** [comment]
p.s. if you are building your own image you will need to pull the fix in efa9395f64190bdb9f1ae38ab0eeb5f7b0d19555, otherwise the automated build on Docker Hub will catch up soon

**Comment by bomeara** [comment]
Thanks, @joshiggins. I should be able to start playing with this later today and I'll report back (finishing some NSF grant stuff at the moment).

**Comment by bomeara** [comment]
I'm playing with this: the need to assign an IP is potentially a problem (I'm within a university environment, so they prob don't like us just picking IPs). I've asked for some static IPs to use.

**Comment by joshiggins** [comment]
Yes, it is annoying if you don't have a network interface you can dedicate to the cluster communication. I'm looking again at Weave to see if there is an alternate solution... let us know how you get on (and hope it isn't too difficult to get those static IPs :wink:)

**Comment by joshiggins** [comment]
@bomeara did you ever get something to work?

**Comment by bomeara** [comment]
No -- investigating again (to see if changes in Docker for Mac have made it easier). Frustratingly, never got straight answer from our IT about static IPs.

**Comment by joshiggins** [comment]
I am starting some changes #5 so that we will not need `--net=host` anymore.
